### PR TITLE
add custom tags property

### PR DESCRIPTION
**Pull request by joejstuart**
This PR adds a property to the publisher that allows for custom tags to be added with any measurement sent with CustomDataMap. This will allow for more flexibility while querying with GROUP BY.

### CODE REVIEW

File: src/main/java/jenkinsci/plugins/influxdb/generators/CustomDataMapPointGenerator.java [hunk 1]
```diff
@@ -28,10 +32,17 @@ public boolean hasReport() {
...
+
         for (String key : customKeys) {
-            Point point = buildPoint(measurementName(key), customPrefix, build)
-                    .fields(customDataMap.get(key))
-                    .build();
+            Point.Builder pointBuilder = buildPoint(measurementName(key), customPrefix, build)
+                    .fields(customDataMap.get(key));
+
+            if (customDataMapTags != null && customDataMapTags.size() > 0) {
+                pointBuilder = pointBuilder.tag(customDataMapTags);
```
Comment (OliverNocon) [inline src/main/java/jenkinsci/plugins/influxdb/generators/CustomDataMapPointGenerator.java]: Currently, this is writing the same tags for all measurements.
I would suggest to do it similar to line 38 and allow different tags per measurement.

Usage would be then like
```
def myCustomTags = [:]
myTags1 = [buildResult: currentBuild.result, NODE_LABELS: env.NODE_LABELS]
myTags2 = [testTag: 'testTag']
myCustomTags["series1"] = myTags1
myCustomTags["series2"] = myTags2
```

This is similar to usage of `customDataMap`: https://github.com/jenkinsci/influxdb-plugin/blob/master/src/main/java/jenkinsci/plugins/influxdb/InfluxDbPublisher.java#L95

Above behavior fits better to `customData`: https://github.com/jenkinsci/influxdb-plugin/blob/master/src/main/java/jenkinsci/plugins/influxdb/InfluxDbPublisher.java#L72

**Review by OliverNocon — COMMENTED** [review]
Thanks this will be very helpful.

The tag functionality has been requested in our use case as well which we have with https://github.com/SAP/jenkins-library/blob/master/vars/influxWriteData.groovy.

I would suggest to handle it similar to `customData` vs. `customDataMap` though and do not force the same tags into all measurements. See my detailed comment. Let me know if you have further questions ...

File: src/main/java/jenkinsci/plugins/influxdb/generators/CustomDataMapPointGenerator.java (hunk 1 above)
Comment (joejstuart) [reply src/main/java/jenkinsci/plugins/influxdb/generators/CustomDataMapPointGenerator.java]: Thanks for the feedback. I agree that would better mirror how cutomDataMap is used. I've updated the code to reflect the changes.

**Comment by OliverNocon** [comment]
Excellent! Looks good from my side.

What about adding the possibility to write tags also for `customData`?

**Comment by joejstuart** [comment]
I just added tags for customData. Please take a look when you can.

**Comment by OliverNocon** [comment]
Thanks very much. To me this looks very good!

**Comment by joejstuart** [comment]
Awesome. Thanks for taking a look.

**Comment by asimell** [comment]
Can you fix the merge conflicts please?

**Pull request merged**